classes/SnippetLoader.php

```diff
@@ -0,0 +1,181 @@
+     *
+     * @param CmsController $cmsController
+     * @param CmsPage $page                 The CMS page for which to load the cache
+     */
+    public static function restoreComponentSnippetsFromCache($cmsController, CmsPage $page)
+    {
+        $componentSnippets = self::fetchCachedSnippets($page);
+
+        foreach ($componentSnippets as $componentInfo) {
+            try {
+                self::attachComponentSnippetToController($componentInfo, $cmsController);
+            } catch (\Exception $e) {
+            }
+        }
+    }
+
+    /**
+     * Attach a component-based snippet to a controller.
+     *
+     * Register the component if it is not defined yet.
+     * This is required because not all component snippets are registered as components,
+     * but it's safe to register them in render-time.
+     *
+     * If asked, the run lifecycle events of the component can be run. This is required for
+     * component that are added late in the page execution like with the twig filter.
+     *
+     * @param array $componentInfo
+     * @param CmsController $controller
+     * @param bool $triggerRun              Should the run events of the component lifecycle be triggered?
+     * @throws SystemException
+     * @throws CmsException
+     */
+    protected static function attachComponentSnippetToController($componentInfo, CmsController $controller, $triggerRun = false)
+    {
+        $componentManager = ComponentManager::instance();
+
+        if (!$componentManager->hasComponent($componentInfo['component'])) {
+            $componentManager->registerComponent($componentInfo['component'], $componentInfo['code']);
+        }
+
+        $component = $controller->addComponent(
+            $componentInfo['component'],
+            $componentInfo['code'],
+            $componentInfo['properties']
+        );
+
+        if ($triggerRun) {
+            if ($component->fireEvent('component.beforeRun', [], true)) {
+                return;
+            }
+
+            if ($component->onRun()) {
+                return;
+            }
+
+            if ($component->fireEvent('component.run', [], true)) {
+                return;
+            }
+        }
+    }
+
+    /**
+     * Store a component snippet to the cache.
+     * The cache is not actually saved; saveCachedSnippets() must be called to persist the cache.
+     *
+     * @param string $alias                     The unique alias of the snippet
+     * @param array $snippetInfo        The info of the snippet
+     */
+    protected static function cacheSnippet($alias, $snippetInfo)
+    {
+        self::$pageSnippetsCache[$alias] = $snippetInfo;
+    }
+
+    /**
+     * Get cached component snippets from the cache.
+     *
+     * @param CmsPage $page         The CMS page for which to load the cache
+     */
+    protected static function fetchCachedSnippets(CmsPage $page)
+    {
+        $cache = @unserialize(Cache::get(self::getMapCacheKey($page), serialize([])));
```

Comment:
Any reason why the error is squelched here? (the `@` symbol)

Reply:
@bennothommo Not sure, I only ported this code from another plugin.

Reply:
@bennothommo any other changes you want here?

Reply:
Yes, I would prefer the error squelching be removed so that the error can be seen if it occurs.

Reply:
Better like this @bennothommo ?